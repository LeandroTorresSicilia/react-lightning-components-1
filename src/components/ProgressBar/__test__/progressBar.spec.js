import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import ProgressBar from './../index';

describe('<ProgressBar/>', () => {
    it('should render a large progress bar', () => {
        const component = renderer.create(<ProgressBar value={40} size="large" />);

        expect(component).toMatchSnapshot();
    });

    it('should render a progress bar with assistive text', () => {
        const component = renderer.create(<ProgressBar assistiveText="Progress: 0%" />);

        expect(component).toMatchSnapshot();
    });

    it('should has set the value passed', () => {
        const component = shallow(<ProgressBar value={25} />);

        expect(component.find('span.slds-progress-bar__value').prop('style').width).toBe('25%');
    });

    it('should be accesible', () => {
        const component = shallow(<ProgressBar value={25} />);

        expect(component.find('div[role="progressbar"]').exists()).toBe(true);
        expect(component.find('div[aria-valuemax="100"]').exists()).toBe(true);
        expect(component.find('div[aria-valuemin="0"]').exists()).toBe(true);
        expect(component.find('div[aria-valuenow=25]').exists()).toBe(true);
    });
});
