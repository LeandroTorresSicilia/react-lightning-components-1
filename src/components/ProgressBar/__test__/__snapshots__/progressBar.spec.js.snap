// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProgressBar/> should render a large progress bar 1`] = `
<div
  aria-valuemax="100"
  aria-valuemin="0"
  aria-valuenow={40}
  className="slds-progress-bar slds-progress-bar_large"
  role="progressbar"
  style={undefined}
>
  <span
    className="slds-progress-bar__value"
    style={
      Object {
        "width": "40%",
      }
    }
  />
</div>
`;

exports[`<ProgressBar/> should render a progress bar with assistive text 1`] = `
<div
  aria-valuemax="100"
  aria-valuemin="0"
  aria-valuenow={0}
  className="slds-progress-bar slds-progress-bar_medium"
  role="progressbar"
  style={undefined}
>
  <span
    className="slds-progress-bar__value"
    style={
      Object {
        "width": "0%",
      }
    }
  >
    <span
      className="slds-assistive-text"
    >
      Progress: 0%
    </span>
  </span>
</div>
`;
